src/tests/ftest/util/spdk_cref.py

```diff
@@ -0,0 +1,417 @@
+class SpdkNvmeCtrlrData(Structure):
+    _fields_ = [
+        ('vid', c_ushort),
+        ('ssvid', c_ushort),
+        ('sn', c_byte * 20),
+        ('mn', c_byte * 40),
+        ('fr', c_ubyte),
+        ('rab', c_ubyte),
+        ('ieee', c_ubyte),
+        ('cmic', Cmic),
+        ('mdts', c_ubyte),
+        ('cntlid', c_ushort),
+        ('ver', SpdkNvmeVsRegister),
+        ('rtd3r', c_uint),
+        ('rtd3e', c_uint),
+        ('oaes', Oaes),
+        ('ctratt', CtrAtt),
+        ('reserved_100', c_ubyte * 12),
+        ('fguid', c_ubyte * 16),
+        ('reserved_128', c_ubyte * 128),
+        ('oacs', Oacs),
+        ('acl', c_ubyte),
+        ('aerl', c_ubyte),
+        ('frmw', Frmw),
+        ('lpa', Lpa),
+        ('elpe', c_ubyte),
+        ('npss', c_ubyte),
+        ('avscc', Avscc),
+        ('apsta', Apsta),
+        ('wctemp', c_ushort),
+        ('cctemp', c_ushort),
+        ('mtfa', c_ushort),
+        ('hmpre', c_uint),
+        ('hmmin', c_uint),
+        ('tnvmcap', c_ulonglong * 2),
+        ('unvmcap', c_ulonglong * 2),
+        ('rpmbs', RpmBs),
+        ('edstt', c_ushort),
+        ('dsto', Dsto),
+        ('fwug', c_ubyte),
+        ('kas', c_ushort),
+        ('mntmt', c_ushort),
+        ('mxtmt', c_ushort),
+        ('sanicap', Sanicap),
+        ('reserved3', c_ubyte * 180),
+        ('sqes', Sqes),
+        ('cqes', Cqes),
+        ('maxcmd', c_ushort),
+        ('nn', c_uint),
+        ('oncs', Oncs),
+        ('fuses', c_ushort),
+        ('fna', Fna),
+        ('vwc', Vwc),
+        ('awun', c_ushort),
+        ('awupf', c_ushort),
+        ('nvscc', c_ubyte),
+        ('reserved531', c_ubyte),
+        ('acwu', c_ushort),
+        ('reserved534', c_ushort),
+        ('sgls', Sgls),
+        ('reserved4', c_ubyte * 228),
+        ('subnqn', c_ubyte * 256),
+        ('reserved5', c_ubyte * 768),
+        ('nvmf_specific', NVMfSpecific),
+        ('psd', SpdkNvmePowerState * 32),
+        ('vs', c_ubyte * 1024)
+    ]
+
+class SpdkNvmeCtrlrOpts(Structure):
+    _fields_ = [
+        ('num_io_queues', c_uint),
+        ('use_cmb_sps', c_bool),
+        ('arb_mechanism', c_uint),
+        ('keep_alive_timeout_ms', c_uint),
+        ('transport_retry_count', c_int),
+        ('io_queue_size', c_uint),
+        ('hostnqn', c_char),
+        ('io_queue_requests', c_uint),
+        ('src_addr', c_char),
+        ('src_svcid', c_char),
+        ('host_id', c_ubyte),
+        ('extended_host_id', c_ubyte),
+        ('command_set', c_uint),
+        ('admin_timeout_ms', c_uint),
+        ('header_digest', c_bool),
+        ('data_digest', c_bool),
+        ('disable_error_logging', c_bool)
+    ]
+
+class SpdkNvmeTransportId(Structure):
+    _fields_ = [
+        ('trtype', c_uint),
+        ('adrfan', c_uint),
+        ('traddr', c_char),
+        ('trsvcid', c_char),
+        ('subnqn', c_char)
+    ]
+
+class SpdkNvmeCtrlr(Structure):
+    pass
+
+class CtrlrEntry(Structure):
+    pass
+
+CtrlrEntry._fields_ = [
```

Comment:
(pylint-protected-access) Access to a protected member _fields_ of a client class